core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1314,11 +1318,24 @@ class Log(@volatile private var _dir: File,
 
   /**
    * Increment the log start offset if the provided offset is larger.
+   *
+   * If the log start offset changed, then this method:
+   *
+   * 1. Records the new log start offset.
+   * 2. Updates the high watermark if it is less than the new log start offset
```

Comment:
Yeah, I agree. I originally decided to add a comment because it was educational to me when I discovered that this call updated a few other components. I changed the documentation to be a bit more high-level and less likely to change.